### PR TITLE
Form styles

### PR DESCRIPTION
**Pull request by mmilo**
this is still a work in progress and needs someone to finish it off, but the ground work is there for making these controls more consistent.

### CODE REVIEW

File: packages/prompt/index.jsx [hunk 1]
```diff
@@ -85,8 +87,15 @@ module.exports = React.createClass({
         cancelText={this.props.cancelText}
         okText={this.props.okText}
         okDisabled={!this.state.valid}>
-        <p>{this.props.content}</p>
-        <input type="text" ref="promptInput" onChange={this.handleValueChange} onKeyUp={this.handleKeyUp} defaultValue={this.props.defaultValue} placeholder={this.props.placeholder} />
+
+        <InputText
+          label={this.props.content}
```
Comment (grapegravity) [inline packages/prompt/index.jsx]: `InputText` only accepts a `string` for this property, while `Prompt` expects a `node`...

Comment (nelix) [inline packages/prompt/index.jsx]: a string is a type of node

Comment (nelix) [inline packages/prompt/index.jsx]: but a node is not a string..

Comment (grapegravity) [inline packages/prompt/index.jsx]: Correct. So this needs revising. Just making notes for my own reference. :smile:

Comment (nelix) [inline packages/prompt/index.jsx]: sorry! :dash:

**Comment by grapegravity** [comment]
This will need to be rebased as the `CancelableEdit` has changed. It’ll also need to take into account the `MarkdownEdit` control.

**Comment by mmilo** [comment]
I think this might be good to go, pending @grapegravity and @nelix’s say so

**Comment by nelix** [comment]
:+1:

File: packages/cancelable-edit/index.jsx [hunk 2]
```diff
@@ -206,31 +209,36 @@ module.exports = React.createClass({
...
 
     if (this.props.creating && !this.state.editing) {
+
       return (
         <Link ref="link" tabIndex="0" href="#" fill={this.props.fill} onClick={this.handleClick}>
           {this.props.createText || this.props.placeholder}
         </Link>
       );
+
     } else {
```
Comment (grapegravity) [inline packages/cancelable-edit/index.jsx]: Extra blank lines are being added around this if/else block, is this intentional?

**Comment by grapegravity** [comment]
Calendars’ focused items don’t have a style, which makes keyboard accessibility not so great (this was an existing issue, maybe we should address it here?)

**Comment by grapegravity** [comment]
The `DeleteButton`'s documentation could do with being updated, as it’s no longer a `button` element.

**Comment by grapegravity** [comment]
Dropdown seems broken?
screen shot 2015-04-27 at 12 24 58 pm

File: packages/form/input-textarea.scss [hunk 3]
```diff
@@ -1,7 +1,10 @@
+@import 'inputs';
+
 .Input.Input--Textarea {
+  @include text-input-style;
   height: auto;
   outline: 0;
   font-weight: 400;
   vertical-align: top;
-  resize: vertical;
+  resize: none;
```
Comment (grapegravity) [inline packages/form/input-textarea.scss]: Are we sure about this?

File: packages/tag-input/index.jsx [hunk 4]
```diff
@@ -1,15 +1 @@
...
-require('./tag-input.scss');
-
-module.exports = React.createClass({
-  displayName: 'TagInput',
-
-  render() {
-    return (<Widgets.Multiselect {...this.props} />);
-  }
-});
+module.exports = require('../form/input-tag');
```
Comment (grapegravity) [inline packages/tag-input/index.jsx]: I feel like adding a deprecation warning here would be useful. Not sure of the best approach, however.

File: style/inputs.scss [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  }
+  &:focus {
+    border-color: $input-state-focus;
+
+    &::placeholder {
+      color: inherit;
+      opacity: .15;
+    }
+  }
+}
```
Comment (grapegravity) [inline style/inputs.scss]: This file has been added, but there’s still the `input-generics` file which contains similar styles?

File: packages/form/input-tag.scss [hunk 6]
```diff
@@ -1,18 +1,174 @@
...
+  &.rw-open {
+    border-bottom-left-radius: 0;
+    border-bottom-right-radius: 0;
+  }
+
+}
+
+.rw-select-list .rw-select-wrapper {
+  border: 0;
 }
```
Comment (grapegravity) [inline packages/form/input-tag.scss]: `InputTag`’s text box has different padding in isolation versus in combination with a calendar component.

File: packages/form/input-textarea.scss (hunk 3 above)
Comment (mmilo) [inline packages/form/input-textarea.scss]: I believe so. The rationale was that we have fixed size textareas that we are intentionally constraining and wouldn’t want to be resized, and we have auto-sizing textareas that will scale to accommodate the content. These two should cover all the bases, but we can build a resize prop to pass into if this is required.

File: packages/form/input-tag.scss (hunk 6 above)
Comment (grapegravity) [inline packages/form/input-tag.scss]: Looks like there are some duplicated/overridden styles which don’t get loaded by `InputTag` on its own.

File: style/inputs.scss (hunk 5 above)
Comment (mmilo) [inline style/inputs.scss]: this is now resolved

**Comment by mmilo** [comment]
I think all raised issues have now been addressed

**Comment by grapegravity** [comment]
The only thing I’d bring up now is that the calendar’s still got some focus issues, but I’m happy to resolve those separately.

**Pull request merged**